Samples/Kaleidoscope/Chapter4/CodeGenerator.cs

```diff
@@ -194,15 +194,15 @@ private Function GetFunction( string name )
 
         private Function GetOrDeclareFunction( Prototype prototype, bool isAnonymous = false )
         {
-            var function = GetFunction( prototype.Identifier.Name );
-            if( function != null )
+            Function retVal = Module.GetFunction( prototype.Identifier.Name );
```

Comment:
What exactly is the issue being fixed here, the commit comments says something about an infinite loop but all this does is rename the variable and remove the "var". The naming etc. as-is was very deliberate to minimize the differences between the chapter variations so that a simple diff between chapters could highlight the new functionality/changes.

Reply:
Never mind I see the difference, missed the **Module.** part of that. I did some tests and made a few additional fixes and applied them to all the chapters samples in (#28) Thanks for finding this.

Reply:
It's ok. Never mind.